src/helper/style-path.js

```diff
@@ -112,40 +120,51 @@ const applyFillColorToSelection = function (colorString, colorIndex, isSolidGrad
         }
 
         // In bitmap mode, fill color applies to the stroke if there is a stroke
-        if (bitmapMode && item.strokeColor !== null && item.strokeWidth) {
-            if (!_colorMatch(item.strokeColor, colorString)) {
-                changed = true;
-                item.strokeColor = colorString;
-            }
-        } else if (isSolidGradient || !item.fillColor || !item.fillColor.gradient ||
-                !item.fillColor.gradient.stops.length === 2) {
+        if (
+            bitmapMode &&
+            !applyToStroke &&
+            item.strokeColor !== null &&
+            item.strokeWidth &&
+            !_colorMatch(item.strokeColor, colorString)
+        ) {
+            changed = true;
+            item.strokeColor = colorString;
+            continue;
+        }
+
+        // TODO: the JS engine may not approve of these dynamic property access shenanigans
```

Comment:
Maybe could clarify that it means the JS performance would be better if we referred to properties by fixed strings